task16/index.js

```diff
@@ -0,0 +1,108 @@
+const URL = "https://myjson.dit.upm.es/api/bins/as8f";
+const tabWrapper = document.getElementById("js-tabWrapper");
+const tab = document.getElementById("js-tabMenu");
+
+const createElementWithClassName = (element, name) => {
+    const createdElement = document.createElement(element);
+    createdElement.classList.add(name);
+    return createdElement;
+};
+
+const addLoading = () => {
+    const loadingPlace = createElementWithClassName('div', "loader");
+    loadingPlace.id = "loadingPlace";
+    tabWrapper.appendChild(loadingPlace);
+}
+
+const removeLoading = () => document.getElementById("loadingPlace").remove();
+
+const getFetchData = async () => {
+    try{
+        const respons = await fetch(URL);
+        if(!respons.ok){
+            throw new Error('ただいまサーバー側で通信が壊れています')
+        };
+        const json = await respons.json();
+        return json;
+    }
+    catch(error) {
+        console.error(error);
+    }
+};
+
+const init = async () => {
+    addLoading();
+    try {
+        const value = await getFetchData();
+        renderTabMenu(value);
+        renderTabContainer(value);
+        // tabSwitch();
+    }
+    catch(error) {
+        console.error(error);
+        tab.textContent = "データを読み込めませんでした";
+    }
+    finally {
+        removeLoading();
+    }
+};
+init();
+
+const renderTabMenu = (value) => {
+    const tabMenuFragment = document.createDocumentFragment();
+    for(let i = 0; i < value.length; i++){
+        const menuList = createElementWithClassName('li', "menu-list");
+        menuList.id = `js-tabMenuList${i}`;
+        menuList.textContent = value[i].category;
+        tabMenuFragment.appendChild(menuList);
+    }
+    tab.appendChild(tabMenuFragment);
+    const addClassActive = document.getElementById("js-tabMenuList0");
+    addClassActive.classList.add("active");
+};
+
+const renderTabContainer = (value) => {
+    const tabContentsFragment = document.createDocumentFragment();
+    for(let i = 0; i < value.length; i++){
+        const newsWrapper = createElementWithClassName('div', "news-wrapper");
+        const newsTitlelist = createElementWithClassName('ul', "news-title-list");
+        const newsImg = createElementWithClassName('img', "news-img")
+        newsWrapper.id = `js-newsWrapper${i}`;
+        newsImg.src = value[i].img;
+
+        tabContentsFragment.appendChild(newsWrapper).appendChild(newsTitlelist).appendChild(getNewsTitle(value[i]));
+
+        tabContentsFragment.appendChild(newsWrapper).appendChild(newsImg);
+    }
+    const tabContainer = createElementWithClassName('div', "tab-container");
+    tabWrapper.appendChild(tabContainer).appendChild(tabContentsFragment);
+    const addClassShow = document.getElementById("js-newsWrapper0");
+    addClassShow.classList.add("show");
+};
+
+const getNewsTitle = ({article}) => {
+    const newsTitleFragment = document.createDocumentFragment();
```

Comment:
IMO
I feel that `create**` is more appropriate.
The reason is that it creates an element and returns it as the return value.

I also thought "article" was better than "news"
Since I received "article" as an argument, I thought it would be more natural to use "article" instead of "news".
```suggestion
const createArticleTitle = ({article}) => {
    const newsTitleFragment = document.createDocumentFragment();
```

Reply:
As you say.
Thank you for pointing that out.
I have also changed the variable name that uses "news" in this function to "article".
https://github.com/seeeen044/JavaScript_handson/commit/64ebfa8ea023bfc87f9916d93d4074b956ca7b60